### ISSUE TITLE
Convert log odds outputs to Probabilities using SHAP Tree explainer for Multiclass classification problems

### ISSUE THREAD

**Issue by geoffkip** [shap/shap#2026]
For Binary classification problems you are able to use the model output parameter (https://shap-lrjball.readthedocs.io/en/docs_update/generated/shap.TreeExplainer.html) to output the classification probabilities instead of the log odds. However, I tried to use this same parameter for a multiclass problem and I get the error below
![image](https://user-images.githubusercontent.com/19884852/120848408-c0e9e700-c542-11eb-9e59-23594da5b29b.png)
My clf XGBoost model object does have a multiclass:softmax objective. Is there any way to output the model probabilities for the different classes in a multiclass model using the SHAP tree explainer?

**Comment by ethankim00** [comment]
This code should work to convert the outputted shap values to the probability space for multiclass problems .

```
def xgb_shap_transform_scale(
    original_shap_values: shap._explanation.Explanation,
    model_prediction: np.ndarraybase_values,
):

    """
    Use modified function posted at
    https://github.com/slundberg/shap/issues/29 to
    transform xgboost SHAP values for classification models from the log odds space to
    the probability space for better interpretability. Uses a simple linear scaling.
    Use the empirical proportions as base value for each class. When calculating on new
    data points, best practice is to use the base values from the original training data
    Args:
        original_shap_values (shap._explnation.Explanation):
            Original SHAP explanation object
        model_prediction (np.ndarray): XGBoost model predictions (in probability space)
        base_values (np.ndarray): Base values for output classes, derived from training
            data (in probability space)
    Returns:
        (shap._explnation.Explanation): Rescaled SHAP values

    """
    # Default to only explain positive class for binary clasification
    if len(original_shap_values.shape) == 2:
        model_prediction = model_prediction[:, 1]
        base_values = base_values[:, 1]
    # Computing the original_explanation_distance
    original_explanation_distance = np.sum(original_shap_values.values, axis=1)
    # Computing the distance between the model_prediction and the transformed base_value
    distance_to_explain = model_prediction - base_values
    # The distance_coefficient is the ratio between both distances
    distance_coefficient = original_explanation_distance / distance_to_explain
    # Transforming the original shapley values to the new scale
    # Iterate over predictors
    if len(original_shap_values.shape) == 2:
        for i in range(original_shap_values.values.shape[1]):
            original_shap_values.values[:, i] = (
                original_shap_values.values[:, i] / distance_coefficient
            )

    else:
        for i in range(original_shap_values.values.shape[1]):
            original_shap_values.values[:, i, :] = (
                original_shap_values.values[:, i, :] / distance_coefficient
            )
    original_shap_values.base_values = base_values

    return original_shap_values
```

**Comment by geoffkip** [comment]
@ethankim00 Thanks Ethan I also created a similar function based on Issue 29. Appreciate you sharing your function as well! Will mark this as closed! Thank you.
